test/framework/src/main/java/org/elasticsearch/index/mapper/FieldMapperTestCase.java

```diff
@@ -0,0 +1,211 @@
+        }),
+        new Modifier("similarity", false, (a, b) -> {
+            a.similarity(new SimilarityProvider("a", new BM25Similarity()));
+            b.similarity(new SimilarityProvider("b", new BM25Similarity()));
+        }),
+        new Modifier("store", supportsStore() == false, (a, b) -> {
+            if (supportsStore()) {
+                a.store(true);
+                b.store(false);
+            }
+        }),
+        new Modifier("term_vector", false, (a, b) -> {
+            a.storeTermVectors(true);
+            b.storeTermVectors(false);
+        }),
+        new Modifier("term_vector_positions", false, (a, b) -> {
+            a.storeTermVectors(true);
+            b.storeTermVectors(true);
+            a.storeTermVectorPositions(true);
+            b.storeTermVectorPositions(false);
+        }),
+        new Modifier("term_vector_payloads", false, (a, b) -> {
+            a.storeTermVectors(true);
+            b.storeTermVectors(true);
+            a.storeTermVectorPositions(true);
+            b.storeTermVectorPositions(true);
+            a.storeTermVectorPayloads(true);
+            b.storeTermVectorPayloads(false);
+        }),
+        new Modifier("term_vector_offsets", false, (a, b) -> {
+            a.storeTermVectors(true);
+            b.storeTermVectors(true);
+            a.storeTermVectorPositions(true);
+            b.storeTermVectorPositions(true);
+            a.storeTermVectorOffsets(true);
+            b.storeTermVectorOffsets(false);
+        })
+    ));
+
+    /**
+     * Add type-specific modifiers for consistency checking.
+     *
+     * This should be called in a {@code @Before} method
+     */
+    protected void addModifier(String property, boolean updateable, BiConsumer<T, T> method) {
+        modifiers.add(new Modifier(property, updateable, method));
+    }
+
+    /**
+     * Add type-specific modifiers for consistency checking.
+     *
+     * This should be called in a {@code @Before} method
+     */
+    protected void addBooleanModifier(String property, boolean updateable, BiConsumer<T, Boolean> method) {
+        modifiers.add(new Modifier(property, updateable, (a, b) -> {
+            method.accept(a, true);
+            method.accept(b, false);
+        }));
+    }
+
+    protected abstract T newBuilder();
+
+    public void testMergeConflicts() {
+        Mapper.BuilderContext context = new Mapper.BuilderContext(SETTINGS, new ContentPath(1));
+        T builder1 = newBuilder();
+        T builder2 = newBuilder();
+        {
+            FieldMapper mapper = (FieldMapper) builder1.build(context);
+            FieldMapper toMerge = (FieldMapper) builder2.build(context);
+            mapper.merge(toMerge);  // identical mappers should merge with no issue
+        }
+        {
+            FieldMapper mapper = (FieldMapper) newBuilder().build(context);
+            FieldMapper toMerge = new FieldMapper("bogus", new MockFieldMapper.FakeFieldType(),
```

Comment:
I think this could just be `FieldMapper toMerge = new MockFieldMapper("bogus")` ?